File: Animal Shelter Opdracht/C# versie/animal_shelter-abstract_interfaces-649f03d658d7546000a2cadc0e118ada4f610607/Animal Shelter/Animals/Dog.cs
using System;

namespace Animal_Shelter.Animals
{
    public class Dog : Animal
    {
        public DateTime LastWalk { get; private set; }
        public bool NeedsWalk
        {
            get
            {
                return (DateTime.Today - this.LastWalk).Days > 0;
            }
        }

        public Dog(string name, Gender gender)
            : base(name, gender)
        {
            this.LastWalk = DateTime.Today;
        }

        public override string ToString()
        {
            return base.ToString() +
                $", last walk: {this.LastWalk.ToShortDateString()}";
        }
    }
}

File: src/Animals/Dog.java
package Animals;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Dog extends Animal implements Observer
{
    DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");

    public Date LastWalk;

    public boolean NeedsWalk;


    public Dog(String name, Animals.Gender gender)
    {
        super(name, gender, 550);
        LastWalk = new Date();
    }

    public void update(Object obj)
    {

            price = price - 50;
            if(price < 50)
            {
                price = 50;
            }

            System.out.println("Name: " + getName() + "      Price: " + price);



    }

    public Date getLastWalk()
    {
        return LastWalk;
    }

    public boolean getNeedsWalk()
    {
        Date today = new Date();
        if(today.after(LastWalk))
        {
            return true;
        }
        return false;
    }

    public String ToString()
    {
        return super.ToString() +
                ", last walk: "+ dateFormat.format(LastWalk) + ", Price: "+ this.price;
    }

}
